src-self-hosted/Module.zig

```diff
@@ -2927,6 +2927,11 @@ pub fn resolvePeerTypes(self: *Module, scope: *Scope, instructions: []*Inst) !Ty
             continue;
         }
 
+        if (prev_inst.ty.zigTypeTag() == .ComptimeInt and next_inst.ty.isInt()) {
```

Comment:
This is missing companion logic for when {int, comptime_int} is encountered rather than {comptime_int, int}